### ISSUE TITLE
Concise display of expanded parent-child hierarchies 

### ISSUE THREAD

**Issue by thohan88** [glin/reactable#147]
First of all, thanks for creating such an amazing package with great documentation. The nice table formatting provided by reactable has encouraged me to transition much of my dashboard work from Power BI/Tableau to Shiny. 

A use case which I am yet to be able to replace, is the concise display of hierarchical data with many metrics. With reactable, a 2-level hierarchy generates two columns with the default settings for `groupBy`. 

![image](https://user-images.githubusercontent.com/16469189/116476619-c2592d00-a87b-11eb-826d-5242d428d5ac.png)

Product hierarchies can often have 3 or more levels where this approach takes up a lot of width in the tables.  Power BI has the ability to expand child rows directly underneath the parent row.

![image](https://user-images.githubusercontent.com/16469189/116476485-9342bb80-a87b-11eb-82c2-f76458a66c80.png)

This leaves more space for adding adding additional columns to the tables.

I have made some attempts to recreate this with reactable, but have not made any significant progress. I would be really grateful to get your first thoughts on whether this is possible with reactable, and how you would approach it.

Thanks again for your awesome work.

**Comment by thohan88** [comment]
Doing a bit more research, I see that this question is similar to the following posts on Stackoverflow:
https://stackoverflow.com/questions/65722241/is-there-a-way-of-making-aggregated-drop-downs-in-reactable-which-doesnt-leave
https://stackoverflow.com/questions/66000137/expand-and-collaps-variables-in-a-single-column-using-reactable-in-r

**Comment by glin** [comment]
This isn't possible today, but similar ideas come up often as feature requests (#123, #112, #33). I would love to make this possible, but I can't say when or if it'll be worked on, unfortunately. These feature requests do help though, so thanks for opening this and linking those SO questions as well.

**Comment by Camil88** [comment]
I encountered exactly the same issue. That would be extremely useful feature. Moreover, that would solve one more issue I'm struggling for some time  - filtering rows in expanded rows. Of course you can use `groupBy` and then filter data in all expanded rows but the problem is as @thohan88 mentioned above - if you have many grouping levels table gets wider and wider and it looks and works not good. I tried to use nested table (in every row) but unfortunately this nested table is not filtered anymore when you search for some value/text (only main table rows are filtered). So it's not useful in my case.

Summing up, if you make this feature possible then we could expand rows and show next grouping levels right under the main rows and additionally we would be also able to filter those expanded rows. That would be stunning enhancement.

**Comment by vkatti** [comment]
You can pass a second level reactable to the details argument of the first reactable as demonstrated [here](https://www.vishalkatti.com/posts/2021-07-27-drilldown/) in my blog.

**Comment by Camil88** [comment]
@vkatti the issue is that you can not filter nested tables using search bar placed in the main table:
![obraz](https://user-images.githubusercontent.com/57962280/134968493-caa42245-360f-4702-b58b-09f123f5fe41.png)
If you type 'Student D' there then no rows are found. The only option is to search data in nested tables' filters (as shown in your example). If toy have any workaround please let us know.
